app/models/concerns/orchestration.rb

```diff
@@ -82,14 +90,26 @@ def record_conflicts
     @record_conflicts ||= []
   end
 
+  def skip_orchestration!
+    @skip_orchestration = true
+  end
+
+  def enable_orchestration!
+    @skip_orchestration = false
+  end
+
+  def skip_orchestration?
+    @skip_orchestration ||= Rails.env == "test"
```

Comment:
this isn't very intuitive. What's the motivation?

On Sunday, June 26, 2016, Shimon Shtein <notifications@github.com> wrote:

> In app/models/concerns/orchestration.rb
> <https://github.com/theforeman/foreman/pull/3574#discussion_r68498935>:
>
> > @@ -82,14 +90,26 @@ def record_conflicts
> >      @record_conflicts ||= []
> >    end
> >
> > +  def skip_orchestration!
> > +    @skip_orchestration = true
> > +  end
> > +
> > +  def enable_orchestration!
> > +    @skip_orchestration = false
> > +  end
> > +
> > +  def skip_orchestration?
> > +    @skip_orchestration ||= Rails.env == "test"
>
> Actually it was intended - in tests I want to skip orchestration, even if
> the code says enable_orchestration!. If a specific test wants the
> orchestration up, it has to stub the method, and force return true, like I
> do in the the test that I have added in this PR.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/theforeman/foreman/pull/3574/files/9f76f0dba6856518562355b69739b502025101ed#r68498935>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALn60ZDWPzg2W0YrjBW8dx7jlsLKtWGks5qPi5zgaJpZM4IvrQ7>
> .
>